sqlglot/dialects/doris.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlglot import exp, generator
+from sqlglot.dialects.dialect import (
+    approx_count_distinct_sql,
+    arrow_json_extract_sql,
+    rename_func,
+)
+from sqlglot.dialects.mysql import MySQL
+from sqlglot.helper import seq_get
+
+
+def _to_date_sql(self: generator.Generator, expression: exp.TsOrDsToDate) -> str:
+    this = self.sql(expression, "this")
+    self.format_time(expression)
+    return f"TO_DATE({this})"
+
+
+def _time_format(
+    self: generator.Generator, expression: exp.UnixToStr | exp.StrToUnix
+) -> t.Optional[str]:
+    time_format = self.format_time(expression)
+    if time_format == Doris.TIME_FORMAT:
+        return None
+    return time_format
+
+
+class Doris(MySQL):
+    DATE_FORMAT = "'yyyy-MM-dd'"
+    DATEINT_FORMAT = "'yyyyMMdd'"
+    TIME_FORMAT = "'yyyy-MM-dd HH:mm:ss'"
+
+    TIME_MAPPING = {
+        "%M": "%B",
+        "%m": "%%-M",
+        "%c": "%-m",
+        "%e": "%-d",
+        "%h": "%I",
+        "%S": "%S",
+        "%u": "%W",
+        "%k": "%-H",
+        "%l": "%-I",
+        "%W": "%a",
+        "%Y": "%Y",
+        "%d": "%%-d",
+        "%H": "%%-H",
+        "%s": "%%-S",
+        "%D": "%%-j",
+        "%a": "%%p",
+        "%y": "%%Y",
+        "%": "%%",
+    }
+
+    class Parser(MySQL.Parser):
+        FUNCTIONS = {
+            **MySQL.Parser.FUNCTIONS,
+            "DATE_TRUNC": lambda args: exp.TimestampTrunc(
+                this=seq_get(args, 1), unit=seq_get(args, 0)
+            ),
+        }
+
+    class Generator(MySQL.Generator):
+        CAST_MAPPING = {}
+
+        TYPE_MAPPING = {
+            **MySQL.Generator.TYPE_MAPPING,
+            exp.DataType.Type.TEXT: "STRING",
+            exp.DataType.Type.TIMESTAMP: "DATETIME",
+            exp.DataType.Type.TIMESTAMPTZ: "DATETIME",
+        }
+
+        TRANSFORMS = {
+            **MySQL.Generator.TRANSFORMS,
+            exp.ApproxDistinct: approx_count_distinct_sql,
+            exp.JSONExtractScalar: arrow_json_extract_sql,
+            exp.JSONExtract: arrow_json_extract_sql,
+            exp.DateDiff: rename_func("DATEDIFF"),
```

Comment:
This should be removed, the correct transform already exists in MySQL's dict:

https://github.com/tobymao/sqlglot/blob/c73790d0249cf0f21f855ed4921c9a8727e2d5ff/sqlglot/dialects/mysql.py#L503

According to Doris' SQL manual you can only have two arguments in `DATEDIFF`, so your override will generate incorrect SQL when you transpile from dialects that can also accept a `unit` argument.

It seems like Starrocks overrides similarly, seems like a bug based on https://docs.starrocks.io/en-us/2.5/sql-reference/sql-functions/date-time-functions/datediff

Reply:
Ok so there's actually a detail I missed: Starrocks supports two variants, called `DATEDIFF` and `DATE_DIFF`. The main difference between is that only the latter supports the `unit` argument, so this needs more careful handling.